lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -541,6 +539,9 @@ export class UserAgentApplication {
 
             // prompt user for interaction
             this.navigateWindow(urlNavigate, popUpWindow);
+
+            const hash = await IframeUtils.monitorWindowForHash(popUpWindow, this.config.system.loadFrameTimeout);
```

Comment:
Won't run for redirect. Can add check for truthiness of `popUpWindow` (and comment).